test/integration/users.test.js

```diff
@@ -1229,4 +1231,80 @@ describe("Users", function () {
         });
     });
   });
+
+  describe("POST /users/migrate", function () {
+    let fetchStub;
+
+    beforeEach(async function () {
+      fetchStub = Sinon.stub(axios, "get");
+    });
+
+    afterEach(async function () {
+      Sinon.restore();
+    });
+
+    it("Should update the user", async function () {
+      fetchStub.resolves({
+        data: githubUserInfo[0]._json,
+      });
+      const usersMigrateResponse = await chai
+        .request(app)
+        .post(`/users/migrate`)
+        .set("Cookie", `${cookieName}=${superUserAuthToken}`);
+      expect(usersMigrateResponse).to.have.status(200);
+      expect(usersMigrateResponse.body).to.eql({
+        message: "Result of migration",
+        data: {
+          totalUsers: 2,
+          usersUpdated: 2,
+          usersNotUpdated: 0,
+          invalidUsersDetails: [],
+        },
+      });
+      const usersResponse = await chai.request(app).get(`/users`).set("cookie", `${cookieName}=${superUserAuthToken}`);
+      expect(usersResponse).to.have.status(200);
+      usersResponse.body.users.forEach((document) => {
+        expect(document).to.have.property(`github_user_id`);
+      });
+    });
+    it("Should return details of users with invalid github username", async function () {
+      fetchStub.rejects({ response: { status: 404 } });
```

Comment:
why are we stubbing 404, isnt that in our control? we usually stub any external calls.

Reply:
Yes, here we are stubbing the external call to GitHub API for fetching the users `github_user_id`